### PR TITLE
feat: codelens to preview manifest

### PR DESCRIPTION
**Pull request by nliu-ms**
Codelens and editor title to preview manifest.  Trigger from manifest template json.
image

### CODE REVIEW

**Comment by nliu-ms** [comment]
E2E TEST: https://github.com/OfficeDev/TeamsFx/blob/b7cf63c1a92f9cbcb0598e013bb81df9118688b4/packages/cli/tests/e2e/smoke/AzureAppHappyPath.tests.ts#L136

File: packages/fx-core/src/core/index.ts [hunk 1]
```diff
@@ -936,6 +936,24 @@ export class FxCore implements Core {
...
   }
 
+  @hooks([
+    ErrorHandlerMW,
+    ConcurrentLockerMW,
+    ProjectSettingsLoaderMW,
+    EnvInfoLoaderMW(true),
+    ContextInjectorMW,
+  ])
+  async getSelectedEnv(
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/src/core/index.ts:946]: What's the return value for non multi env?

**Review by jayzhang — DISMISSED** [review]

File: packages/fx-core/src/core/index.ts (hunk 1 above)
Comment (nliu-ms) [reply packages/fx-core/src/core/index.ts:946]: Thanks! Update to throw error for non multi-env.

File: packages/vscode-extension/src/codeLensProvider.ts [hunk 2]
```diff
@@ -89,3 +89,33 @@ export class AdaptiveCardCodeLensProvider implements vscode.CodeLensProvider {
...
+    if (matches != null) {
+      const match = matches[0];
+      const line = document.lineAt(document.positionAt(matches.index).line);
+      const indexOf = line.text.indexOf(match);
+      const position = new vscode.Position(line.lineNumber, indexOf);
+      const range = new vscode.Range(
+        position,
+        new vscode.Position(line.lineNumber, indexOf + match.length)
+      );
+      const command = {
```
[the remark this answers is not in the document]
Comment (nliu-ms) [reply packages/vscode-extension/src/codeLensProvider.ts:111]: Cause I want to put it just on the top of schema. Will add another button like `Open schema` later.

[the remark this answers is not in the document]
Comment (HuihuiWu-Microsoft) [reply packages/vscode-extension/src/codeLensProvider.ts:111]: So there'll be two buttons: 'openPreviewFile' and 'openSchema'. Shouldn't the button 'openSchema' be on the top of schema and 'openPreviewFile' on top of the whole file?

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**